### PR TITLE
Don't use DebugPrint like sequence for char[].

### PR DESCRIPTION
**Pull request by vng**
Follow up https://github.com/organicmaps/organicmaps/pull/7316

### CODE REVIEW

File: base/internal/message.hpp [hunk 1]
```diff
@@ -132,7 +132,11 @@ std::string inline DebugPrint(std::nullopt_t const & p)
   return "nullopt";
 }
 
-template <typename T, size_t N> inline std::string DebugPrint(T (&arr) [N])
+// Avoid calling it for string literals.
+template <typename T, size_t N,
+         typename = std::enable_if_t<!std::is_same<typename std::remove_cv<T>::type, char>::value &&
+                                     !std::is_same<typename std::remove_cv<T>::type, char16_t>::value>>
```
Comment (biodranik) [inline base/internal/message.hpp:138]: Would it be better if our implementation overloads and prints these characters, by converting them into the utf-8 and using std::string?

Comment (vng) [reply base/internal/message.hpp:138]: And it already does via other existing overloads. What happens with char16_t in C++20 will be decided after C++20.
I don't see that we need it now.

**Comment by vng** [comment]
UPD: I added tests to eliminate doubts.

File: base/internal/message.hpp (hunk 1 above)
Comment (biodranik) [reply base/internal/message.hpp:129]: Is it still needed with https://github.com/organicmaps/organicmaps/pull/7324 merged? I would prefer to get build errors when someone uses it inappropriately.

**Comment by vng** [comment]
- Yes, it is needed.
- I have added tests that are not working properly in master.
- Check my added tests. Should they work ok or "compile error"?

File: base/internal/message.hpp (hunk 1 above)
Comment (biodranik) [reply base/internal/message.hpp:129]: Can you please explain the intention here?

Comment (vng) [reply base/internal/message.hpp:129]: Print char[] like string "foo", not like sequence of chars "[3: f, o, o ]"

Comment (biodranik) [reply base/internal/message.hpp:129]: It may be useful to know that an array is printed. Maybe ["foo"] or "foo"[] ?

**Pull request merged**